src/containers/Registration/Registration.tsx

```diff
@@ -55,11 +60,15 @@ const Registration: React.FC = () => {
   const { login } = useAuth();
   const history = useHistory();
 
+  const checkFromValidation = () => {
+    return values.validUsername && values.validEmail && values.validPassword;
+  };
+
   const handleSubmit = () => {
     const username = usernameRef.current?.value?.toLowerCase();
     const password = passwordRef.current?.value;
     const email = emailRef.current?.value;
-    if (username && password) {
+    if (username && password && checkFromValidation()) {
```

Comment:
If we implement `isValid` (`checkFormValidation`) correctly, we don't have to check for `username` and `password` anymore.